### PR TITLE
chore(cli): remove dead code

### PR DESCRIPTION
**Pull request by kitsonk**
I noticed that several modules in `cli/main.rs` where `pub` when we don't currently expect anyone to consume the cli crate at the moment.  Removing those public modules resulted in exposing a lot of dead code that was be obfuscated by the public modules.  That code has been removed in this PR.

### CODE REVIEW

File: cli/media_type.rs [hunk 1]
```diff
@@ -75,30 +73,6 @@ impl MediaType {
...
-
-  /// Convert a MediaType to a `ts.Extension`.
-  ///
-  /// *NOTE* This is defined in TypeScript as a string based enum.  Changes to
-  /// that enum in TypeScript should be reflected here.
-  pub fn as_ts_extension(&self) -> &str {
-    match self {
-      MediaType::JavaScript => ".js",
-      MediaType::JSX => ".jsx",
-      MediaType::TypeScript => ".ts",
```
Comment (bartlomieju) [inline cli/media_type.rs:87]: Isn't this code used in new module graph/new tsc?

File: cli/ops/io.rs [hunk 2]
```diff
@@ -184,14 +183,12 @@ impl StreamResourceHolder {
 }
 
 pub enum StreamResource {
-  Stdin(tokio::io::Stdin, TTYMetadata),
```
Comment (bartlomieju) [inline cli/ops/io.rs:187]: This is surprising...

File: cli/worker.rs [hunk 3]
```diff
@@ -194,21 +194,6 @@ impl Worker {
     self.js_runtime.mod_evaluate(id)
   }
 
-  /// Loads, instantiates and executes provided source code
-  /// as module.
-  pub async fn execute_module_from_code(
-    &mut self,
-    module_specifier: &ModuleSpecifier,
-    code: String,
-  ) -> Result<(), AnyError> {
```
Comment (bartlomieju) [inline cli/worker.rs:203]: I'd prefer to leave this method - I'm debugging test runner and there's a good chance this API will be needed.

**Review by bartlomieju — APPROVED** [review]
Actually LGTM, let's land it, I can bring code back as needed

**Comment by ry** [comment]
Nice work!

**Pull request merged**